core/src/query.rs

```diff
@@ -131,6 +137,14 @@ impl MongoStatement for MongoQuery {
         // If the query timeout is 0, it means "no timeout"
         let options = if self.query_timeout.is_some_and(|timeout| timeout > 0) {
             opt.max_time(Duration::from_millis(self.query_timeout.unwrap() as u64))
+                .batch_size({
+                    // Natacha and I aren't actually sure what the default batch size is, but we think it might be like 101.
+                    if rowset_size > 101 {
+                        rowset_size as u32
+                    } else {
+                        101
+                    }
+                })
```

Comment:
The idea here is to only update the batch size when `rowset_size` is greater than the default batch size. Otherwise, we may update the batch size to be a very small number.

Reply:
Yes, although, I couldn't find that default anywhere in the code.
But here is the doc : [https://www.mongodb.com/docs/drivers/rust/current/fundamentals/crud/read-operations/retrieve/#modify-aggregation-behavior](url)
I think that not setting any value if it's below a threshold that we deem suitable, then it will be better.
I don't think that "forcing" the default when not needed is necessary. Especially since we are not sure the default value we use here is 100% correct.

Reply:
Yeah good point. Do we know what the threshold should be?